test/shopify-cli/commands/help_test.rb

```diff
@@ -25,10 +41,47 @@ def test_default_behavior_lists_tasks
         end
         output = io.join
 
-        assert_match('Available commands', output)
+        assert_match('Available core commands:', output)
         assert_match(/Usage: .*shopify/, output)
       end
 
+      def test_local_commands_available_within_a_project
+        Project.stubs(:current_project_type).returns('rails')
+        Registry.add(->() { Rails::Commands::Fake }, 'fake_rails')
+
+        io = capture_io do
+          run_cmd('help')
+        end
+        output = io.join
+
+        assert_match(/Available commands for Ruby on Rails App projects.*fake_rails/m, output)
```

Comment:
I'm using `/m` to match across multiple lines, but this isn't ideal, I'd rather test against the structure than the presentation. Hopefully we can find a way to untangle these later.